go/libraries/doltcore/sqle/tables.go

```diff
@@ -1507,7 +1595,20 @@ func (t *AlterableDoltTable) updateFromRoot(ctx *sql.Context, root *doltdb.RootV
 }
 
 func (t *AlterableDoltTable) CreateCheck(ctx *sql.Context, check *sql.CheckDefinition) error {
-	sch := t.sch
+	root, err := t.getRoot(ctx)
+	if err != nil {
+		return err
+	}
+
+	updatedTable, _, err := root.GetTable(ctx, t.tableName)
+	if err != nil {
+		return err
+	}
+
+	sch, err := updatedTable.GetSchema(ctx)
```

Comment:
i'm not sure i understand why we're going table -> root -> schema in some of these cases

Reply:
ALTER TABLE statements can have any number of clauses, which means that CreateCheck could be called after AddColumn for example. `t.sch` is out of date in this case (having been set at table creation time). So we need to get the updated version of the table and schema in the session root.